script/initdb

```diff
@@ -55,10 +57,16 @@ if ($help) {
     print "  --init_database : Use the generated deployment files created with --prepare_init\n";
     print "                    to actually initialize a database.\n";
     print "  --force         : Force overwriting existing data.\n";
+    print "  --user=login    : Change uid before connecting the DB.\n";
     print "  --help          : This help message.\n";
     exit;
 }
 
+if ($user) {
+    my $uid = getpwnam($user) || die "No such login $user";
+    setuid($uid) || die "can't suid to $user";
```

Comment:
Any reason not to also call setgid? When passing user names one would normally also even call initgroups to set the supplementary groups.